tests/visualization_tests/test_optimization_history.py

```diff
@@ -209,11 +215,77 @@ def objective(trial: Trial) -> float:
     for x, study in zip(suggested_params, studies):
         study.enqueue_trial({"x": x})
         study.optimize(objective, n_trials=1)
-    figure = plot_optimization_history(studies, error_bar=True)
-
-    mean = np.mean(suggested_params)
-    std = np.std(suggested_params)
-
-    np.testing.assert_almost_equal(figure.data[0].y, mean)
-    np.testing.assert_almost_equal(figure.data[2].y, mean + std)
-    np.testing.assert_almost_equal(figure.data[3].y, mean - std)
+    info_list = _get_optimization_history_info_list(
+        studies, target=None, target_name="Objective Value", error_bar=True
+    )
+    mean = np.mean(suggested_params).item()
+    std = np.std(suggested_params).item()
+    assert info_list == [
+        _OptimizationHistoryInfo(
+            [0],
+            _ValuesInfo([mean], [std], "Objective Value"),
+            _ValuesInfo([mean], [std], "Best Value"),
+        )
+    ]
+
+
+optimization_history_info_lists = [
+    [],  # Empty info.
+    [  # Vanilla.
+        _OptimizationHistoryInfo(
+            [0, 1, 2],
+            _ValuesInfo([1.0, 2.0, 0.0], None, "Dummy"),
+            None,
+        )
+    ],
+    [  # Multiple.
+        _OptimizationHistoryInfo(
+            [0, 1, 2],
+            _ValuesInfo([1.0, 2.0, 0.0], None, "Dummy"),
+            None,
+        ),
+        _OptimizationHistoryInfo(
+            [0, 1, 2],
+            _ValuesInfo([1.0, 1.0, 1.0], None, "Dummy"),
+            None,
+        ),
+    ],
+    [  # With best values.
+        _OptimizationHistoryInfo(
+            [0, 1, 2],
+            _ValuesInfo([1.0, 2.0, 0.0], None, "Dummy"),
+            _ValuesInfo([1.0, 1.0, 1.0], None, "Best Value"),
+        )
+    ],
+    [  # With error bar.
+        _OptimizationHistoryInfo(
+            [0, 1, 2],
+            _ValuesInfo([1.0, 2.0, 0.0], [1.0, 2.0, 0.0], "Dummy"),
+            None,
+        )
+    ],
+    [  # With best values and error bar.
+        _OptimizationHistoryInfo(
+            [0, 1, 2],
+            _ValuesInfo([1.0, 2.0, 0.0], [1.0, 2.0, 0.0], "Dummy"),
+            _ValuesInfo([1.0, 1.0, 1.0], [1.0, 2.0, 0.0], "Best Value"),
+        )
+    ],
+]
+
+
+@pytest.mark.parametrize("target_name", ["Objective Value", "Target Name"])
+@pytest.mark.parametrize("info_list", optimization_history_info_lists)
+def test_get_optimization_history_plot(
+    target_name: str, info_list: List[_OptimizationHistoryInfo]
+) -> None:
+    figure = _get_optimization_history_plot(info_list, target_name=target_name)
+    assert figure.layout.yaxis.title.text == target_name
+    expected_legends = []
+    for info in info_list:
+        expected_legends.append(info.values_info.label_name)
+        if info.best_values_info is not None:
+            expected_legends.append(info.best_values_info.label_name)
+    legends = [scatter.name for scatter in figure.data if scatter.name is not None]
+    assert sorted(legends) == sorted(expected_legends)
+    figure.write_image(BytesIO())
```

Comment:
How about removing line, which aims to check the figure successfully drawn, and defining a test independently like https://github.com/optuna/optuna/blob/master/tests/visualization_tests/test_contour.py#L150-L161
```suggestion
```

Reply:
Do you suggest this from the perspective of readability?

Reply:
Could be? I suggested it for consistency with the other test for visualization such as this part in [`test_contour.py`](https://github.com/optuna/optuna/blob/7bebfd657eb33c0d0a3219e2307901b9b85e642d/tests/visualization_tests/test_contour.py#L150-L161), which I linked above.